Bfo css backgrounds fix

faceless2 opened the pull request:
Tests were referencing the wrong reference test

<!-- Reviewable:start -->

<!-- Reviewable:end -->

xiuqijix committed:
Add a DataTransfer test

 The types attribute must return this DataTransfer object's types array

FremyCompany committed:
[css-fonts-4] Contribute Microsoft tests for Font Variations

FremyCompany committed:
Address Sergey's feedback (part 1)

FremyCompany committed:
Address Sergey's feedback (part 2)

FremyCompany committed:
More linting fixup

FremyCompany committed:
Added missing element

FremyCompany committed:
Fixed incorrect calc() tests -- values should be clamped not rejected

FremyCompany committed:
Addressed @drott's feedback

FremyCompany committed:
Fixed lint issues

FremyCompany committed:
Remove fallback to linked font in installed-font test cases

FremyCompany committed:
Fixed some duplicate test names

csnardi committed:
Rewrite CSSOM index tests

index-001.html and index-002.html had nondescript names, and index-002.html used generate_tests. Merge index-001.html into shorthand-serialization.html, and create shorthand-values.html from index-002.html.

Builds off the work of #5331.

csnardi committed:
Correct serialization of overflow

Commit:
[svg 2 text] test for length and indexed access

rsolomakhin committed:
Replace 'enabledMethods' array into 'method' string.

darrnshn committed:
[css-typed-om] Stub list-valued tests.

Currently we're just adding a TODO when we want to test a list-valued
property. It's probably easier if we stub that test function and then
implement it later.

Bug: 774887
Change-Id: Iab3cea915afbcc8490cacbd45e5632a18e7dece1
Reviewed-on: https://chromium-review.googlesource.com/981956
Reviewed-by: Rune Lillesveen <futhark@chromium.org>
Commit-Queue: Darren Shen <shend@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546067}

Commit:
Tests were referencing incorrect reference test

Commit:
Tests were referencing incorrect reference test

Commit:
Tests were referencing incorrect reference test

Commit:
Merge branch 'bfo-css-backgrounds-fix' of https://github.com/faceless2/web-platform-tests into bfo-css-backgrounds-fix

w3c-bots commented:
# Build PENDING

Started: None
Finished: None


<h2>Failing Jobs</h2>
<ul>

<li>chrome:dev</li>

</ul>



<h2>Unstable Results</h2>
  
  
  
  <h3>Browser: "Chrome Dev"<small></small></h3>
  <p>View in: <a href="http://pulls.web-platform-tests.org/job/26496.5">WPT PR Status</a> |
      <a href="https://travis-ci.org/w3c/web-platform-tests/jobs/358799045">TravisCI</a></p>
  <table>
    <tr>
      <th>Test</th>
      <th>Subtest</th>
      <th>Results</th>
      <th>Messages</th>
    </tr>
  
  
  <tr>
    <td><code>/css/css-backgrounds/css3-background-origin-padding-box.html</code></td>
    <td>&nbsp;</td>
    <td>PASS: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  <tr>
    <td><code>/css/css-backgrounds/css3-background-size-001.html</code></td>
    <td>&nbsp;</td>
    <td>PASS: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  </table>

faceless2 commented:
Hello reviewers. This should be a very uncontentious change but I can't seem to get any love from travis. Is it me?

lilles committed:
[css-typed-om] Add support for the column-width property.

Bug: 820299
Change-Id: I9e380416b18f30294fe7cb969a0ce7d0ab1fd7bd
Reviewed-on: https://chromium-review.googlesource.com/977910
Reviewed-by: Darren Shen <shend@chromium.org>
Commit-Queue: Rune Lillesveen <futhark@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546082}

rsolomakhin committed:
Merge pull request #10180 from rsolomakhin/method

Replace 'enabledMethods' array with 'method' string.

stephenmcgruer committed:
Port nested sticky tests from reftests to JS tests

Bug: 699244
Change-Id: Ia85a44be9d9f7b0d0db382892520f2755523f514
Reviewed-on: https://chromium-review.googlesource.com/980236
Reviewed-by: Robert Flack <flackr@chromium.org>
Commit-Queue: Stephen McGruer <smcgruer@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546172}

nhiroki committed:
Worker: Add tests for import.meta.url on DedicatedWorker

Bug: 680046
Change-Id: Ieb341c5922ad247961ca477893037896dbd88ea6
Reviewed-on: https://chromium-review.googlesource.com/974726
Commit-Queue: Hiroki Nakagawa <nhiroki@chromium.org>
Reviewed-by: Matt Falkenhagen <falken@chromium.org>
Reviewed-by: Kouhei Ueno <kouhei@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546254}

hwanseung committed:
[css-typed-om] support r and rx and ry properties

r[1], rx[2], ry[3]
are added to support in whitelist.
and test file are also added.

[1]https://svgwg.org/svg2-draft/geometry.html#R
[2]https://svgwg.org/svg2-draft/geometry.html#RX
[3]https://svgwg.org/svg2-draft/geometry.html#RY

Bug: 820299
Change-Id: I51457648ca07aaf06cf15bd28452d5ad8099d2e3
Reviewed-on: https://chromium-review.googlesource.com/981918
Commit-Queue: Hwanseung Lee <hwanseung@chromium.org>
Reviewed-by: Darren Shen <shend@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546302}

darrnshn committed:
[css-typed-om] Support some text-decoration properties.

Adds tests for:
- text-decoration
- text-decoration-line
- text-decoration-skip (not implemented in Blink)
- text-decoration-skip-ink
- text-decoration-width (not implemented in Blink)

text-decoration-line uses a weird internal structure for CSSValues,
so we have to encode/decode to that structure when getting/setting.
Failing tests because for some reason we ignore 'blink' for the
computed value. Blink also doesn't support 'spelling-error' and
'grammar-error' yet.

Bug: 820299
Change-Id: I60e8c3958c9b3a784177c9813a7325aa26e39581
Reviewed-on: https://chromium-review.googlesource.com/981717
Commit-Queue: Darren Shen <shend@chromium.org>
Reviewed-by: Rune Lillesveen <futhark@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546364}

kinu committed:
Re-land: "Don't adjust the NavigationTimings on redirects"

The reason for revert was the new navigation timing test expectations
(that got commited right before the original change) had a FAIL
expectation for the particular event order that this change was fixing.

Original change's description:
> Don't adjust the NavigationTimings on redirects
>
> We adjust the timings in WebDocumentLoaderImpl::UpdateNavigation(),
> shouldn't update in each AddRedirect() (which is called in a batch
> way after all the redirects are handled in the browser process with PlzNavigate,
> adjusting timings there with current timestamp is totally wrong)
>
> R=​ksakamoto,arthursonzogni
>
> Bug: 813889
> Change-Id: I3a57d3fdf1833c300feb5ee61737b64cece0946b
> Reviewed-on: https://chromium-review.googlesource.com/974673
> Reviewed-by: Kunihiko Sakamoto <ksakamoto@chromium.org>
> Commit-Queue: Kinuko Yasuda <kinuko@chromium.org>
> Cr-Commit-Position: refs/heads/master@{#546001}

Bug: 813889
Change-Id: I6f5c0ba1196fa1dd52225036820b29a95569ac21
Reviewed-on: https://chromium-review.googlesource.com/981913
Reviewed-by: Kunihiko Sakamoto <ksakamoto@chromium.org>
Commit-Queue: Kinuko Yasuda <kinuko@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546365}

marcoscaceres committed:
fix: assert_equals check is wrong

marcoscaceres committed:
Add @edenchuang again 🎊

bfgeek committed:
[css-layout-api] Implementation of LayoutConstraints#fixedInlineSize.

This adds the bare bones for the LayoutConstraints object. It just
handles the fixedInlineSize parameter. I expect the information in
this object after the next F2F.

The layout for the test will simply output an autoBlockSize of 100, if
the fixedInlineSize is 100. The tests impose various conditions in
how this can occur, but is certainly not exhaustive.

Change-Id: I6be4a7b77d7b737c778dc2de8b235035b8d3fa22
Bug: 726125
Reviewed-on: https://chromium-review.googlesource.com/981517
Commit-Queue: Ian Kilpatrick <ikilpatrick@chromium.org>
Reviewed-by: Morten Stenshorne <mstensho@chromium.org>
Reviewed-by: Emil A Eklund <eae@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546393}

marcoscaceres committed:
fix copy/pasta

hwanseung committed:
[css-typed-om] support x, y, cx, cy properties

x[1], y[2], cx[3], cy[4]
are added to support in whitelist.
and test file are also added.

[1]https://svgwg.org/svg2-draft/geometry.html#X
[2]https://svgwg.org/svg2-draft/geometry.html#Y
[3]https://svgwg.org/svg2-draft/geometry.html#CX
[4]https://svgwg.org/svg2-draft/geometry.html#CY

Bug: 820299
Change-Id: I46ced4427a5e7b2e08aa132fc0b8c5d2e1f19d22
Reviewed-on: https://chromium-review.googlesource.com/983077
Commit-Queue: Hwanseung Lee <hwanseung@chromium.org>
Reviewed-by: Darren Shen <shend@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546426}

kritisingh1 committed:
Test that event handlers set on the window object are found on the body element (#10169)

Based on event-handler-attributes-body-window.html.

hayatoito committed:
Keep track of all pending slot assignments, and recalc all before UpdateStyle phase

For the purpose of rendering, Blink must calculate all pending slot assignments
before UpdateStyle phase. Unless that, a recursive RecalcStyle() may not
traverse a node which should be re-attached.

e.g. Suppose the following tree, where #d1 is assigned to a slot.

host
├──/shadow-root
│   └── slot name=s1
└── div id=d1 slot=s1

Then, we change #d1's slot attribute to s2, like:

> document.querySelector('#d1').setAttribute('slot', 's2');

host
├──/shadow-root
│   └── slot name=s1
└── div id=d1 slot=s2

In this case, #d1 should be removed from a LayoutTree. In other words, Blink
has to reattach #d1 somehow. However, if we don't recalc a slot assignment for
the shadow tree before UpdateStyle, a recursive RecalcStyle never traverses the
sub-tree because child-needs-style-recalc flag is not set for the node (and its
ancestor nodes). A flag should be set as a result of slot assignment recalc.

Thanks to the Incremental Shadow DOM, a slot assignemt recalc now becomes a
local operatoin on each shadow tree, rather than one global operation for every
shadow trees. We don't need to traverse a composed tree to recalc all. We can
recalc a slot assignment for each shadow tree directly, without traversing a
composed tree.

For a shadow tree which is not connected, we don't need to recalc its slot
assignment before UpdateStyle because such a shadow tree shouldn't have any
effect on rendering. Lazy slot assignment recalc is enough for such a shadow
tree.

This CL doesn't introduce any optimization to minimize the number of
to-be-reattached nodes. I'll optimize that as a next task. I'll use a sort of
dynamic programming there, as I did at
https://chromium-review.googlesource.com/c/chromium/src/+/535493 for
non-Incremental Shadow DOM.

Except for the performance, this CL should be the last part of Incremental
Shadow DOM from the external behavior's perspective. Style and Layout should
work correctly after this CL, even with Incremental Shadow DOM.

BUG=776656

Change-Id: Id18e87ff59d92863c68c571e7db09253c08aa91f
Reviewed-on: https://chromium-review.googlesource.com/964062
Reviewed-by: Takayoshi Kochi <kochi@chromium.org>
Reviewed-by: Rune Lillesveen <futhark@chromium.org>
Commit-Queue: Hayato Ito <hayato@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546458}

annevk committed:
DOM: reset target/relatedTarget

For https://github.com/whatwg/dom/pull/585.

rsolomakhin committed:
Merge pull request #10206 from w3c/marcoscaceres-patch-2

fix copy/pasta

rsolomakhin committed:
Merge pull request #10204 from w3c/marcoscaceres-patch-1

Add @edenchuang again 🎊

rsolomakhin committed:
Merge pull request #10203 from w3c/fixup_show_test

fix: assert_equals check is wrong

henbos committed:
RTCRtpSender.getStats() in blink added behind flag.

This exposes RTCRtpSender.getStats() in JavaScript (behind flag) which
implements the stats selection algorithm[1] for senders.

[1] https://w3c.github.io/webrtc-pc/#dfn-stats-selection-algorithm

Bug: 680172
Change-Id: I8117c87f475d1c78fa3301fc2d821f0c3a21281f
Reviewed-on: https://chromium-review.googlesource.com/975306
Commit-Queue: Henrik Boström <hbos@chromium.org>
Reviewed-by: Philip Jägenstedt <foolip@chromium.org>
Reviewed-by: Harald Alvestrand <hta@chromium.org>
Reviewed-by: Taylor Brandstetter <deadbeef@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546493}

odejesush committed:
bluetooth: FakeBluetoothChooser impl.

Implements the interfaces that FakeBluetoothChooser overrides. The
mojom::FakeBluetoothChooser overrides are implemented by running
|event_handler_| with the appropriate arguments. The BluetoothChooser
overrides do nothing at the moment, since the base class is only meant
to update the chooser user interface.

This change also begins to partially convert a Bluetooth chooser test to test
the interface and deletes the temporary tests that were created to test this
API.

BUG=719827,719826,824704

Change-Id: I56a04e6bdb2de38d9163d7bc350db578deeb9ba3
Reviewed-on: https://chromium-review.googlesource.com/962983
Reviewed-by: Daniel Cheng <dcheng@chromium.org>
Reviewed-by: Conley Owens <cco3@chromium.org>
Reviewed-by: Giovanni Ortuño Urquidi <ortuno@chromium.org>
Commit-Queue: Ovidio Henriquez <odejesush@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546509}

whimboo committed:
Use detailed failure message for wdspec test failures (pytest).

When adding failures to the test report, the hook for pytest_runtest_logreport
missed to extract the failure message from the stack, and as such passed in no
message for the failure.
bugzilla-url: https://bugzilla.mozilla.org/show_bug.cgi?id=1433923
gecko-commit: 849020c741de1e3bca2e22f5902e7d3b40f64ebd
gecko-integration-branch: central
gecko-reviewers: jgraham

whimboo committed:
Find element for (partial) link text has to use rendered content.

To retrieve links via "link text" or "partial link text" the rendered
content of the element has to be used. This can be the case for CSS
transformations like "uppercase".
bugzilla-url: https://bugzilla.mozilla.org/show_bug.cgi?id=1381519
gecko-commit: 3e204686f1b10441f48435890241dff6706d04dd
gecko-integration-branch: central
gecko-reviewers: ato

jgraham committed:
fixup! Fix indentation

bzbarsky committed:
part 1.  Fix the buggy existing load-event test for stylesheets.

The test was adding the load listener to the <link> _after_ the load event on
the <link> had already fired.
bugzilla-url: https://bugzilla.mozilla.org/show_bug.cgi?id=1443652
gecko-commit: ba6cef216a171782bd8abcd00a08c9ae88cbfe43
gecko-integration-branch: central
gecko-reviewers: bholley

bzbarsky committed:
part 2.  Add a bunch of web platform tests for load and error events on stylesheet links.
bugzilla-url: https://bugzilla.mozilla.org/show_bug.cgi?id=1443652
gecko-commit: 7daaf289c085e05e0e4a0327a13748002fb467a0
gecko-integration-branch: central
gecko-reviewers: bholley

lukebjerring committed:
Handle idlharness.js expected errors as an IdlArray.assert_throw option (#10164)

* Handle idlharness.js expected errors as an IdlArray.assert_throw option

wanderview committed:
P2 Add a WPT test verifying synthetic responses work when a request fragment is present.
bugzilla-url: https://bugzilla.mozilla.org/show_bug.cgi?id=1443850
gecko-commit: 7b66eb1629750401eb5768646d8381ff4e631cac
gecko-integration-branch: central
gecko-reviewers: asuth

fdoray committed:
Revert "bluetooth: FakeBluetoothChooser impl."

This reverts commit 188adfb51a7c4ca1ab0cbeaffdec4f7cd76857f7.

Reason for revert: https://crbug.com/826787

Original change's description:
> bluetooth: FakeBluetoothChooser impl.
>
> Implements the interfaces that FakeBluetoothChooser overrides. The
> mojom::FakeBluetoothChooser overrides are implemented by running
> |event_handler_| with the appropriate arguments. The BluetoothChooser
> overrides do nothing at the moment, since the base class is only meant
> to update the chooser user interface.
>
> This change also begins to partially convert a Bluetooth chooser test to test
> the interface and deletes the temporary tests that were created to test this
> API.
>
> BUG=719827,719826,824704
>
> Change-Id: I56a04e6bdb2de38d9163d7bc350db578deeb9ba3
> Reviewed-on: https://chromium-review.googlesource.com/962983
> Reviewed-by: Daniel Cheng <dcheng@chromium.org>
> Reviewed-by: Conley Owens <cco3@chromium.org>
> Reviewed-by: Giovanni Ortuño Urquidi <ortuno@chromium.org>
> Commit-Queue: Ovidio Henriquez <odejesush@chromium.org>
> Cr-Commit-Position: refs/heads/master@{#546509}

TBR=dcheng@chromium.org,cco3@chromium.org,ortuno@chromium.org,odejesush@chromium.org

Change-Id: I0597151ec5878d41ac631b72296203bc99b1f2c8
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Bug: 719827, 719826, 824704, 826787
Reviewed-on: https://chromium-review.googlesource.com/984299
Commit-Queue: François Doray <fdoray@chromium.org>
Reviewed-by: François Doray <fdoray@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546572}

lukebjerring committed:
Support partial dictionaries (#10159)

* Support partial dictionaries

darrnshn committed:
[css-typed-om] Add support for marker properties.

This patch adds support for marker, marker-start, marker-end, marker-mid
We also had to add a new CSS data type <url>, but because it's not
spec'd yet, we can't really test it.

Bug: 820299
Change-Id: Ie3f851be3d65396705594d99582ac3e87e3fc57d
Reviewed-on: https://chromium-review.googlesource.com/983192
Reviewed-by: Rune Lillesveen <futhark@chromium.org>
Commit-Queue: Darren Shen <shend@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546645}

burg committed:
webdriver: add more WebKit workarounds to detect whether a window is fullscreen (#10226)

WebKit doesn't implement the Fullscreen API yet, so we need to use
the vendor prefixed property to check whether the window is fullscreen.

This workaround was previously applied to fullscreen_window.py, but it
needs to happen in other places as wel.

ZhuoyuQian committed:
Check BackgroundEdgeOrigin when check BackgroundPosition.

We should check Background{X/Y}Origin when check
BackgroundPosition{X/Y}. BackgroundPosition same but
BackgroundEdgeOrigin different is not the same position, we should
return false when FillLayersEqual().

Bug: 823046

Signed-off-by: Zhuoyu Qian <zhuoyu.qian@samsung.com>
Change-Id: Ia7d0c5af33e1ee5986420dc054c2ce1b1f65e9b0
Reviewed-on: https://chromium-review.googlesource.com/967912
Reviewed-by: Rune Lillesveen <futhark@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546679}

odejesush committed:
bluetooth: FakeBluetoothChooser impl.

Implements the interfaces that FakeBluetoothChooser overrides. The
mojom::FakeBluetoothChooser overrides are implemented by running
|event_handler_| with the appropriate arguments. The BluetoothChooser
overrides do nothing at the moment, since the base class is only meant
to update the chooser user interface.

This change also begins to partially convert a Bluetooth chooser test to test
the interface and deletes the temporary tests that were created to test this
API.

This change reverts commit 42624492cd5e80855f1dd3a5a8146b258def2c2a, which
reverted the original change. The original change caused the
new-scan-device-added.html test to fail because of a Mojo validation error
resulting from not updating the
third_party/.../external/wpt/resources/chromium/fake_bluetooth_chooser.mojom.js
file needed by the test.

Bug: 719827, 719826, 824704, 826787
Change-Id: I5e5cce7e809b4d71216c2d074a417ea822d5e527
Reviewed-on: https://chromium-review.googlesource.com/985195
Commit-Queue: Ovidio Henriquez <odejesush@chromium.org>
Reviewed-by: Giovanni Ortuño Urquidi <ortuno@chromium.org>
Reviewed-by: Daniel Cheng <dcheng@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546729}

frivoal committed:
[css-aling] tests for row-gap / column-gap normal

Matching the https://github.com/w3c/csswg-drafts/issues/2294 spec change

Not including tests for used value in multicol, because (so far) it is
UA defined.

bashi committed:
Remove create_file_system_file()

This function isn't used.

Bug: 658997,402387
Change-Id: I01d751dfebaafd1580beaeb12ebcd124df132b26
Reviewed-on: https://chromium-review.googlesource.com/984919
Commit-Queue: Kenichi Ishibashi <bashi@chromium.org>
Reviewed-by: Matt Falkenhagen <falken@chromium.org>
Reviewed-by: Kinuko Yasuda <kinuko@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546765}

henbos committed:
RTCRtpReceiver.getStats() in blink added behind flag.

This exposes RTCRtpReceiver.getStats() in JavaScript (behind flag) which
implements the stats selection algorithm[1] for receivers.

(This is similar to RTCRtpSender.getStats():
https://chromium-review.googlesource.com/c/chromium/src/+/975306)

[1] https://w3c.github.io/webrtc-pc/#dfn-stats-selection-algorithm

Bug: 680172
Change-Id: I8049a52fcaa3c2bd51e5541c7149d9b3aee57e3d
Reviewed-on: https://chromium-review.googlesource.com/976121
Commit-Queue: Henrik Boström <hbos@chromium.org>
Reviewed-by: Taylor Brandstetter <deadbeef@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546801}

foolip committed:
Invert (fix) the order of `wpt make-hosts-file` instructions (#10237)

This was inverted so that the Windows instructions were printed on
Linux. Rather than just changing the condition, switch the messages,
to match the order of some preceding code.

frivoal committed:
[css-overflow] Add the spec editors as OWNERS (#10234)

fdoray committed:
Revert "bluetooth: FakeBluetoothChooser impl."

This reverts commit 6a337f19713e049cd42969b0c909c59aff4146cd.

Reason for revert: https://crbug.com/824704

These tests are failing:

external/wpt/bluetooth/server/getPrimaryServices/blocklisted-services-with-uuid.https.html
external/wpt/bluetooth/server/getPrimaryServices/blocklisted-services.https.html
external/wpt/bluetooth/service/getCharacteristic/gen-blocklisted-characteristic.https.html
external/wpt/bluetooth/service/getCharacteristics/blocklisted-characteristics.https.html
external/wpt/bluetooth/service/getCharacteristics/gen-blocklisted-characteristic-with-uuid.https.html

Original change's description:
> bluetooth: FakeBluetoothChooser impl.
>
> Implements the interfaces that FakeBluetoothChooser overrides. The
> mojom::FakeBluetoothChooser overrides are implemented by running
> |event_handler_| with the appropriate arguments. The BluetoothChooser
> overrides do nothing at the moment, since the base class is only meant
> to update the chooser user interface.
>
> This change also begins to partially convert a Bluetooth chooser test to test
> the interface and deletes the temporary tests that were created to test this
> API.
>
> This change reverts commit 42624492cd5e80855f1dd3a5a8146b258def2c2a, which
> reverted the original change. The original change caused the
> new-scan-device-added.html test to fail because of a Mojo validation error
> resulting from not updating the
> third_party/.../external/wpt/resources/chromium/fake_bluetooth_chooser.mojom.js
> file needed by the test.
>
> Bug: 719827, 719826, 824704, 826787
> Change-Id: I5e5cce7e809b4d71216c2d074a417ea822d5e527
> Reviewed-on: https://chromium-review.googlesource.com/985195
> Commit-Queue: Ovidio Henriquez <odejesush@chromium.org>
> Reviewed-by: Giovanni Ortuño Urquidi <ortuno@chromium.org>
> Reviewed-by: Daniel Cheng <dcheng@chromium.org>
> Cr-Commit-Position: refs/heads/master@{#546729}

TBR=dcheng@chromium.org,cco3@chromium.org,ortuno@chromium.org,odejesush@chromium.org

Change-Id: Ie2ef2af1ae74e8dd5ca01441027adaa59e6e0c0f
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Bug: 719827, 719826, 824704, 826787
Reviewed-on: https://chromium-review.googlesource.com/986592
Reviewed-by: François Doray <fdoray@chromium.org>
Commit-Queue: François Doray <fdoray@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546851}

frivoal committed:
[css-overflow-3] test the flow-relative overflow properties (#10233)

Corresponding to the spec changes decided in https://github.com/w3c/csswg-drafts/issues/2000

lukebjerring committed:
Update the webrtc-pc IDL file (#9853)

* Update webrtc-pc IDL

lukebjerring committed:
Update the geometry IDL file (#9787)

* Update geometry IDL
* Prevent multiple testing of the same interfaces.

Commit:
Update the name of webaudio to web-audio-api

lukebjerring committed:
Update the webauthn IDL file (#9850)

* Add webauthn idl file
* Simplify webauthn/interfaces.https.html test

lukebjerring committed:
Check inherited interfaces exist (#10240)

* Check inherited interfaces exist
* Re-jig IdlHarness error capture + report ordering

dtapuska committed:
Add wpt test for back/forward mouse buttons.

Test that back and forward mouseup events are received and can be
preventDefaulted.

BUG=680741

Change-Id: I381a3c3dacc344ae4f49c8a35cdcc0ef83907917
Reviewed-on: https://chromium-review.googlesource.com/986508
Reviewed-by: Navid Zolghadr <nzolghadr@chromium.org>
Commit-Queue: Dave Tapuska <dtapuska@chromium.org>
Cr-Commit-Position: refs/heads/master@{#546857}

frivoal committed:
[css-align] Fix typo and clean-up markup

frivoal committed:
Merge pull request #10232 from frivoal/gap-normal

[css-align] tests for row-gap / column-gap normal

tanderson-google committed:
Replace Arial with Arimo GCS font

BUG=787020
TBR=dpranke,asvitkine

Change-Id: I38740d1ee9fe621eabd5416f1a87b23a71fbb7e7
Reviewed-on: https://chromium-review.googlesource.com/986812
Reviewed-by: Thomas Anderson <thomasanderson@chromium.org>
Commit-Queue: Thomas Anderson <thomasanderson@chromium.org>
Cr-Commit-Position: refs/heads/master@{#547246}

lukebjerring committed:
Revert #10240

eholk committed:
Delete wasm/many-memories.window.js WPT test

Testing for a specific number is causing problems and determining the right
number is problematic in general, as running out of memory is always a
legitimate possiblity. See https://github.com/WebAssembly/design/issues/1167 for
more discussion. In light of this, it's best to just remove the test.

Bug:
Change-Id: I66e248c8ab851bca0c54fa6fe13b6e93e55f1264
Reviewed-on: https://chromium-review.googlesource.com/851081
Commit-Queue: Brad Nelson <bradnelson@chromium.org>
Reviewed-by: Brad Nelson <bradnelson@chromium.org>
Cr-Commit-Position: refs/heads/master@{#547333}

svgeesus committed:
Merge pull request #10172 from liamquin/svg-lists-have-length

[svg 2 text] add scripted tests for length and indexed access

horo-t committed:
Set SkipServiceWorker flag for synchronous loads from the main thread.

Before 5e1b52dd7e828fb2e4bd69f599c0c30eec3e873c, synchronous XHR on worker
was handled by service workers. It is because the |is_sync_load| was false
when the sync request is from worker thread. But after the CL, the
|is_sync_load| flag for the sync request from worker became true, so the request
will not go to the service worker.

This CL will fix this by
 - Set the SkipServiceWorker flag for synchronous loads from the main thread
   in the renderer process. (FetchParameters.cpp)
 - Don't set skip_service_worker even if is_sync_load is true in the browser
   process. (resource_dispatcher_host_impl.cc)

Bug: 706331,827473
Change-Id: I186bc97f3f8d298e0a04942d0ec4b708b3022cc1
Reviewed-on: https://chromium-review.googlesource.com/989376
Reviewed-by: Kinuko Yasuda <kinuko@chromium.org>
Commit-Queue: Tsuyoshi Horo <horo@chromium.org>
Cr-Commit-Position: refs/heads/master@{#547407}

mrego committed:
[css-grid] Fix resolution of percentage paddings and margins of grid items

We were not resolving properly percentage paddings and margins
for tracks that have something like minmax(auto, 100px).
The reason was that while computing the minimum size of a grid item,
the percentages were resolved against the inline size of the grid container.
But for grid items we shouldn't never use the grid container size,
but the grid area size, as that's their containing block.

The patch modifies ContainingBlockLogicalWidthForContent() and
ContainingBlockLogicalHeightForContent() in LayoutBox,
so for grid items we return 0 if the area size hasn't been set yet.
We never want to use the grid container's sizes in these cases.

BUG=808758
TEST=external/wpt/css/css-grid/grid-items/grid-items-percentage-margins-*
TEST=external/wpt/css/css-grid/grid-items/grid-items-percentage-paddings-*

Change-Id: Ib142e51aee1fe623d38688469b179f01f82eb07b
Reviewed-on: https://chromium-review.googlesource.com/980756
Reviewed-by: Javier Fernandez <jfernandez@igalia.com>
Commit-Queue: Manuel Rego Casasnovas <rego@igalia.com>
Cr-Commit-Position: refs/heads/master@{#547417}

ricea committed:
ReadableStream: getReader() should call ToString() on mode

When the "mode" option is not undefined, getReader({mode}) algorithm
calls ToString() on it before checking whether or not it is "blob". Add
a test to ensure this conversion is done.

csnardi committed:
Update IRC archive link (#10266)

The link was dead; update to the new link.

tkent-google committed:
custom-elements: Add tests for callback timing with HTML parser.

Bug: 821831
Change-Id: I5b3e1978e37ce34e6fb885ca1077ea90ab2f494c
Reviewed-on: https://chromium-review.googlesource.com/987972
Reviewed-by: Takayoshi Kochi <kochi@chromium.org>
Commit-Queue: Kent Tamura <tkent@chromium.org>
Cr-Commit-Position: refs/heads/master@{#547424}

csnardi committed:
Remove test that CSSStyleSheet.media is read-only

style-sheet-interfaces-001.html tested that styleSheet.media was
read-only, but the spec changed to add a [PutForwards] attribute to
|media|, meanining that the attribute is no longer truly read-only.
assert_readonly can have side-effects if the property has a
[PutForwards] attribute, so remove this check, as it was causing the
next test to fail.

Bug: 821780
Change-Id: I77bee2f11484f6e679e4aee6b0641179c686d1d3
Reviewed-on: https://chromium-review.googlesource.com/989407
Reviewed-by: Emil A Eklund <eae@chromium.org>
Commit-Queue: Chris Nardi <cnardi@chromium.org>
Cr-Commit-Position: refs/heads/master@{#547451}

anforowicz committed:
Tests and docs for the current CORB vs <link rel="preload"> behavior.

Bug: 809261
Cq-Include-Trybots: master.tryserver.chromium.linux:linux_mojo
Change-Id: Idafb2b4cc5da3d6c7a054e479745288cc3b86513
Reviewed-on: https://chromium-review.googlesource.com/984338
Commit-Queue: Łukasz Anforowicz <lukasza@chromium.org>
Reviewed-by: Nick Carter <nick@chromium.org>
Cr-Commit-Position: refs/heads/master@{#547494}

inexorabletash committed:
Cookie Store API: Split out and enable test cases

Split out cases from a master "cookie-store-tests.js" suite,
allowing them to be enabled/debugged separately.

Also updates observation tests to use the new "change" event,
and deletes "static" since we can assume a dependency on a test
server here.

Bug: 783229
Change-Id: I9e29ea74a2f0513a033a8f9c71da2836c000ee27
Reviewed-on: https://chromium-review.googlesource.com/986649
Commit-Queue: Joshua Bell <jsbell@chromium.org>
Reviewed-by: Victor Costan <pwnall@chromium.org>
Cr-Commit-Position: refs/heads/master@{#547543}

anforowicz committed:
Make CORB MIME type classification consistent with the web specs.

https://tools.ietf.org/html/rfc7303 says that if "new media type is
introduced for an XML-based format, the name of the media type SHOULD
end with '+xml'".

https://tools.ietf.org/html/rfc6839 covers '+xml' and '+json' suffixes.

https://mimesniff.spec.whatwg.org/#xml-mime-type says "An XML MIME type
is any MIME type whose subtype ends in '+xml' or whose essence is
'text/xml' or 'application/xml'. [RFC7303]".

https://mimesniff.spec.whatwg.org/#json-mime-type says "A JSON MIME type
is any MIME type whose subtype ends in '+json' or whose essence is
'application/json' or 'text/json'."

There are no occurences of "application/xml+", "text/xml+",
"application/json+", "text/json+" or "text/x-json" in the specs above
and on various lists of MIME types like:
-
https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Complete_list_of_MIME_types
- https://en.wikipedia.org/wiki/Media_type
- https://www.freeformatter.com/mime-types-list.html
- https://www.sitepoint.com/mime-types-complete-list/

Bug: 826756
Cq-Include-Trybots: master.tryserver.chromium.linux:linux_mojo
Change-Id: Ied30f9728bd4f082bb620fea150f342457ea4833
Reviewed-on: https://chromium-review.googlesource.com/985211
Commit-Queue: Łukasz Anforowicz <lukasza@chromium.org>
Reviewed-by: Nick Carter <nick@chromium.org>
Cr-Commit-Position: refs/heads/master@{#547565}

domenic committed:
Fix customElements.upgrade() tests for <template> (#10272)

* Fix customElements.upgrade() tests for <template>

See https://github.com/w3c/web-platform-tests/pull/9869#issuecomment-378002725.

* Actually fix things

odejesush committed:
bluetooth: web: test: Update blocklist UUID tests

This change updates the blocklisted UUID tests to use the Web
Bluetooth Test API. This will allow the test to be migrated into wpt/.

BUG=509038

Change-Id: I72acd10e5aabc4c03f955dc4c7b2a8620930cc2d
Reviewed-on: https://chromium-review.googlesource.com/894738
Commit-Queue: Ovidio Henriquez <odejesush@chromium.org>
Reviewed-by: Giovanni Ortuño Urquidi <ortuno@chromium.org>
Cr-Commit-Position: refs/heads/master@{#547588}

csnardi committed:
Serialize font shorthand like any other shorthand

The font shorthand had a short-circuit in StylePropertySerializer, which
meant that it was never serialized as a shorthand for style.cssText,
citing reasons of web incompatibility. However, it appears that all
other major browsers serialize font as a shorthand instead of having
style.cssText expand to all properties. Remove this short-circuit to
match other browsers.

Bug: 827921
Change-Id: I4d522b41a6153808b6b8dab17831afdd8315fdc3
Reviewed-on: https://chromium-review.googlesource.com/989834
Commit-Queue: Chris Nardi <cnardi@chromium.org>
Reviewed-by: Timothy Loh <timloh@chromium.org>
Cr-Commit-Position: refs/heads/master@{#547607}

darrnshn committed:
[css-typed-om] Add a few keyword properties.

Add support for the following properties:
- image-rendering
- text-anchor
- text-combine-upright
- text-overflow
- text-rendering
- transform-style
- user-select

Lots of failing tests, usually because a keyword isn't implemented in
Blink yet.

Bug: 820299
Change-Id: Ifd0193ac55d12a31d28b486166ad78a38012314b
Reviewed-on: https://chromium-review.googlesource.com/985337
Reviewed-by: Rune Lillesveen <futhark@chromium.org>
Commit-Queue: Darren Shen <shend@chromium.org>
Cr-Commit-Position: refs/heads/master@{#547665}

Commit:
Throw IdlHarnessError outside try

heycam committed:
fix reftest-wait typo (#10276)

bzbarsky committed:
Use stable error messages in the link onload event tests by default.

alvestrand committed:
Don't enforce name rule for RTCDTMFToneChangeEvent

Since the constructor of RTCDTMFToneChangeEvent is exposed,
creating such events with other names than "tonechange" is possible.
No reason to discriminate against such.

Also adds tests for constructor.

Bug: chromium:822799
Change-Id: I4b36f3094acee200dd4200c4d24f6b46e10a06e6
Reviewed-on: https://chromium-review.googlesource.com/992038
Reviewed-by: Guido Urdaneta <guidou@chromium.org>
Commit-Queue: Harald Alvestrand <hta@chromium.org>
Cr-Commit-Position: refs/heads/master@{#547742}

bzbarsky committed:
Merge pull request #10286 from bzbarsky/stable-error-messages-for-link-load

Use stable error messages in the link onload event tests by default.

svgeesus committed:
Merge pull request #9373 from w3c/user/frremy/contribute-css-fonts-4-tests

[css-fonts-4] Contribute Microsoft tests for Font Variations

lukebjerring committed:
Updated ServiceWorker IDL file (#9886)

* Updated ServiceWorker IDL file

Commit:
Automated Keyboard Input (#10113)

lukebjerring committed:
Updated filter-effects IDL file (#9884)

* Updated filter-effects IDL file

mkruisselbrink committed:
[Mojo Blob URLs] Resolve navigation blob URLs in blink.

This makes sure that no race conditions exist between navigating to a
blob URL and revoking it. Currently only implemented for the network
service code path.

Also add one more navigation-then-revoke WPT test, as the existing tests
didn't cover the NavigationScheduler code-path.

Design doc: https://docs.google.com/document/d/1DqVcTWE9Qb_3KpIRH2bFV-6hWEr8S92c4ppY67YL1KI/edit#heading=h.n4fzuc6mgziv

Further navigation cases are handled in https://chromium-review.googlesource.com/c/chromium/src/+/899508.

Bug: 800901
Cq-Include-Trybots: master.tryserver.chromium.linux:linux_mojo
Change-Id: Icf9a4725526b4329a2841626977292802ad2fa7a
Reviewed-on: https://chromium-review.googlesource.com/898009
Reviewed-by: Daniel Cheng <dcheng@chromium.org>
Reviewed-by: Kinuko Yasuda <kinuko@chromium.org>
Reviewed-by: Nasko Oskov <nasko@chromium.org>
Commit-Queue: Marijn Kruisselbrink <mek@chromium.org>
Cr-Commit-Position: refs/heads/master@{#547842}

jugglinmike committed:
Correct reference to certificate file

kojiishi committed:
Fix a typo in wpt/css/css-text-decor/text-emphasis-style-001.html

This test does not pass because of a typo in the ref file.

Change-Id: I2673137789267359544a50e126d3ba16b728770b
Reviewed-on: https://chromium-review.googlesource.com/989455
Commit-Queue: Philip Jägenstedt <foolip@chromium.org>
Reviewed-by: Philip Jägenstedt <foolip@chromium.org>
Cr-Commit-Position: refs/heads/master@{#547847}

foolip committed:
Document useful arguments for `wpt run chrome` (#10238)

darrnshn committed:
[css-typed-om] Support animation and transition properties.

This patch supports all the animation/transition properties except
animation-name and transition-property (need spec feedback first).
We also fix the <time> tests so that computed time values use canonical
units as spec'd in [1].

[1] https://github.com/w3c/css-houdini-drafts/issues/725

Bug: 820299
Change-Id: I7c038ce419f8bdb1c789493554776c5e7f9810c7
Reviewed-on: https://chromium-review.googlesource.com/983094
Commit-Queue: Darren Shen <shend@chromium.org>
Reviewed-by: Rune Lillesveen <futhark@chromium.org>
Cr-Commit-Position: refs/heads/master@{#548017}

Commit:
Merge branch 'bfo-css-backgrounds-fix' of https://github.com/faceless2/web-platform-tests into bfo-css-backgrounds-fix

faceless2 commented:
Argh. Sorry. Done.
